### PR TITLE
feat: update file permission on commit

### PR DESCRIPTION
**Pull request by mshima**

### CODE REVIEW

**Comment by mshima** [comment]
Clover report is broken at Istanbul, took quite a while to discover 😞.

File: __tests__/commit-file-async.js [hunk 1]
```diff
@@ -6,14 +6,19 @@ const path = require('path');
...
 const sinon = require('sinon');
 const editor = require('..');
+const { STATE, STATE_MODIFIED, STATE_DELETED } = require('../lib/state');
 
 const rmSync = filesystem.rmSync || filesystem.rmdirSync;
 
-describe('#commit()', () => {
-  const outputRoot = path.join(os.tmpdir(), 'mem-fs-editor-test');
+// Permission mode are handled differently by windows, skip it.
+const posixIt = process.platform === 'win32' ? it.skip : it;
```
Comment (SBoudrias) [inline __tests__/commit-file-async.js:14]: I'd recommend adding windows to the matrix; seems like it'd bring value for this package specifically :)

**Review by SBoudrias — APPROVED** [review]
LGTM, I'm letting you merge once you're ready.

File: __tests__/commit-file-async.js (hunk 1 above)
Comment (SBoudrias) [reply __tests__/commit-file-async.js]: Not sure if this was clear, but I meant in the github action; so we run the UTs in windows.

Comment (mshima) [reply __tests__/commit-file-async.js]: It’s already running.
Or I am missing something?

Comment (SBoudrias) [reply __tests__/commit-file-async.js]: oh lol 🤦🏻 you're right, this is already the case!

**Pull request merged**